### ISSUE TITLE
Color picker doesn't use correct cursor position when placed inside popover

### ISSUE THREAD

**Issue by a-dpq** [tovic/color-picker#27]
I'm trying to use the color picker inside a Bootstrap popover, but when I do this, the color picker doesn't use the correct cursor position when selecting a color. It behaves as if the color picker and the cursor were in the upper left of the screen. You can see this behavior by clicking on any of the color swatches on [this page](http://depasquale.art/bug-demo). Is there any way I can get the color picker to use the correct cursor position?

**Comment by tovic** [comment]
Make sure that your `<html>` and `<body>` element has been set as `position:static;` in the CSS.

**Comment by a-dpq** [comment]
I tried that just now, but the result is the same. I updated the [page](http://a-dpq.github.io/bug-demo) so that you can see for yourself.

**Comment by tovic** [comment]
Current color picker version does not detect the color picker pane placement to determine the relative cursor position based on its parent(s) container offset. This enhancement is achieve-able but too complicated for something that is &lsquo;simple&rsquo;. Most cursor position will break when the color picker pane is placed inside an element with `position` other than `static`. I see that you use `picker.enter(popOverElement)` here, which by default has been set to `position:absolute;`.

**Comment by tovic** [comment]
Don&rsquo;t be too dependent with that **X** framework. Simply use CSS alone to style the color picker appearance so that it becomes a _Bootstrap_&rsquo;s pop-over widget.

~~~ .css
.color-picker-control {
  background:black;
  border:0;
  padding:15px;
  margin-top:15px;
  border-radius:4px;
  position:relative;
}
.color-picker-control:before {
  content:"";
  display:block;
  width:0;
  height:0;
  border:10px solid transparent;
  border-bottom-color:black;
  position:absolute;
  bottom:100%;
  left:10px;
}
~~~

**Comment by a-dpq** [comment]
Thanks for your feedback, though I'm still not sure how to solve this problem. I tried pasting the CSS into the page, but it still doesn't work. Maybe I'll just try a different color picker solution if there is no easy fix for my page.

**Comment by tovic** [comment]
Remove your `index.js` file then do a standard `let foo = new CP(bar)` to instantiate the color picker for each color box trigger. Let that CSS alone modify the appearance, no custom markup required here (except for that pre-defined color box).

**Comment by a-dpq** [comment]
Sorry, but I still don't understand. The entire logic for my site is included in the index.js file, so I can't remove it.

**Comment by flamestream** [comment]
I believe @a-dpq is talking about the issue of when the container with the trigger is inside an overflowed panel. There is currently a bug with the way the offset is calculated in that instance.

This is apparent on https://tovic.github.io/color-picker/color-picker.fit.html when executing the following code and testing the library when clicking an item after scrolling down.
```JavaScript
var s = document.querySelectorAll('p')[1].style; s.height = '300px'; s.overflow = 'auto';
```
I've submitted a pull request that fixes the issue at #28.

**Comment by tovic** [comment]
#29
